content/courses/advanced-ada/parts/resource_management/memory_management.rst

```diff
@@ -263,15 +263,91 @@ number of components.
 Storage elements
 ----------------
 
-.. admonition:: Relevant topics
+We saw parts of the :ada:`System.Storage_Elements` package while discussing
+about :ref:`addresses <Adv_Ada_Addresses>`. However, we haven't discussed yet
+the main types from that package: :ada:`Storage_Element` and
+:ada:`Storage_Array`.
 
-    - :ada:`System.Storage_Elements` package
-    - :ada:`System.Storage_Elements.Storage_Element'Size`
-        - :arm22:`The Package System.Storage_Elements <13-7-1>`
+We defined :ref:`storage elements <Adv_Ada_Storage_Elements>` previously.
+In the :ada:`System.Storage_Elements` package, a storage element is represented
+by the :ada:`Storage_Element` type. Its size (:ada:`Storage_Element'Size`) is
+equal to :ada:`Storage_Unit` |mdash| which we also mentioned previously.
 
-.. todo::
+The :ada:`Storage_Array` type is an array type of storage elements. This is its
+definition:
 
-    Complete section!
+.. code-block:: ada
+
+    type Storage_Array is
+      array (Storage_Offset range <>) of
+        aliased Storage_Element;
+
+A storage array is used to represent a contiguous sequence of storage elements
+in memory. In other words, you can think of an object of :ada:`Storage_Array`
+type as a (memory) buffer.
+
+.. admonition:: For further reading
+
+    Note that the :ada:`Storage_Offset` is a range type. It's used not only
```

Comment:
Should "range type" be "integer type"?  I'm not familiar with the term "range type" anyway...

Reply:
Yes, indeed, I've just changed it.